src/server/game/Spells/SpellEffects.cpp

```diff
@@ -2232,6 +2232,15 @@ void Spell::EffectSummonType(SpellEffIndex effIndex)
             break;
     }
 
+    // Some creatures must be summoned as pets, yet have a different category or type set in SummonProperties.dbc.
+    // Until a generic way is found to handle them, replace the existing values.
+    switch (properties->Id)
+    {
+    case 628: // Hungry Plaguehound
+        properties->Category = SUMMON_CATEGORY_PET;
```

Comment:
Better get an idea about the TrinityCore threading model (WordUpdateLoop, Map::Update, MapInstance::Update, WorldSession::Update) to ensure no race conditions are added. In this case it seems you are modifying a singleton field every time Spell::EffectSummonType() is called for a particular summon, technically triggering a race condition if it is possible for multiple threads to call this function at same time (leaving aside for a moment the technicality of 2 threads writing the exact same value).